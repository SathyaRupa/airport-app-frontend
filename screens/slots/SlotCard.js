import * as React from 'react';
import {StyleSheet, TouchableOpacity, View} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import {Card, Text} from 'react-native-paper';
import UpdateButton from '../../components/UpdateButton';
import DeleteButton from '../../components/DeleteButton';

function SlotCard({
  id,
  value,
  icon,
  handleDelete,
  handleUpdate,
  onPress,
  testId,
}) {
  const SlotText = ({text}) => {
    return (
      <View>
        <Text variant="bodyMedium" style={styles.text} numberOfLines={1}>
          {text}
        </Text>
      </View>
    );
  };
  return (
    <View style={styles.mainCard}>
      <TouchableOpacity onPress={onPress} testID={testId}>
        <Card style={styles.slotCard}>
          <LinearGradient
            colors={['#4D869C', '#7AB2B2']}
            start={{x: 0, y: 0}}
            end={{x: 1, y: 1}}
            style={styles.gradient}>
            <View style={styles.cardContent}>
              <Card.Content>
                <View style={styles.content}>
                  <View style={styles.icon}>{icon}</View>
                  <View style={styles.textContent}>
                    <SlotText text={`Start: ${value.startTime}`} />
                    <SlotText text={`End: ${value.endTime}`} />
                    <SlotText text={`${value.available}`} />
                  </View>
                </View>
              </Card.Content>
              <Card.Actions>
                <View style={styles.buttonContainer}>
                  <UpdateButton onPress={() => handleUpdate(id, value)} />
                  <DeleteButton onPress={() => handleDelete(id, value)} />
                </View>
              </Card.Actions>
            </View>
          </LinearGradient>
        </Card>
      </TouchableOpacity>
    </View>
  );
}
export default SlotCard;

const styles = StyleSheet.create({
  mainCard: {
    padding: 7,
  },
  slotCard: {
    backgroundColor: '#4D869C',
  },
  text: {
    color: 'white',
    padding: 2,
    marginRight: 0,
    fontWeight: 'bold',
  },
  content: {
    flex: 3,
    height: 100,
    alignContent: 'flex-start',
    flexDirection: 'row',
  },
  textContent: {
    flexDirection: 'column',
    padding: 15,
    width: 300,
  },
  cardContent: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  buttonContainer: {
    flex: 2,
    flexDirection: 'row',
    marginLeft: -114,
  },
  gradient: {
    borderRadius: 12,
    overflow: 'hidden',
  },
  icon: {
    paddingTop: 25,
  },
});
