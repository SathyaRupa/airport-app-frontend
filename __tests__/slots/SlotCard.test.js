import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import SlotCard from '../../screens/slots/SlotCard';

jest.mock('react-native-linear-gradient', () => 'LinearGradient');
jest.mock('react-native-toast-message', () => 'Toast');

jest.useFakeTimers();

const slot = {
  id: '1',
  startTime: '2024-05-11T07:19:27.265575Z',
  endTime: '2024-05-31T13:08:03.019702Z',
  status: 'Booked',
};

describe('SlotCard Component', () => {
  it('snapshot test', () => {
    const slotCard = render(<SlotCard id={1} value={slot} />);
    expect(SlotCard).toMatchSnapshot();
  });
  it('should be able to click on delete icon', () => {
    const mockHandleDelete = jest.fn();
    const slotCard = render(
      <SlotCard
        id={1}
        value={slot}
        icon="icon"
        handleDelete={mockHandleDelete}
      />,
    );
    const deleteIcon = slotCard.queryByTestId('delete-icon');
    expect(deleteIcon).toBeTruthy();
    fireEvent.press(deleteIcon);
    expect(mockHandleDelete).toHaveBeenCalledWith(1, slot);
  });

  it('should be able to click on update icon', () => {
    const mockHandleUpdate = jest.fn();
    const slotCard = render(
      <SlotCard
        id={1}
        value={slot}
        icon="icon"
        handleUpdate={mockHandleUpdate}
      />,
    );
    const updateIcon = slotCard.queryByTestId('update-icon');
    expect(updateIcon).toBeTruthy();
    fireEvent.press(updateIcon);
    expect(mockHandleUpdate).toHaveBeenCalledWith(1, slot);
  });

  it('SlotCard should be clickable', () => {
    const mockHandleOnPress = jest.fn();
    const {queryByTestId} = render(
      <SlotCard
        id={1}
        value={slot}
        icon="icon"
        onPress={mockHandleOnPress}
        testId="slot-card-0"
      />,
    );
    const slotCard = queryByTestId('slot-card-0');
    expect(SlotCard).toBeTruthy();
    fireEvent.press(slotCard);
    expect(mockHandleOnPress).toHaveBeenCalledTimes(1);
  });
});
