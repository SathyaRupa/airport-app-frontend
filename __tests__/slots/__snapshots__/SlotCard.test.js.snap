// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotCard Component snapshot test 1`] = `[Function]`;
